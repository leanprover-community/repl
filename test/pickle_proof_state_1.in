{"cmd" : "import Lean"}

{"cmd" : "def f : Nat := by sorry", "env": 0, "sorries": "individual"}

{"pickleTo": "test/c.olean", "proofState": 0}

{"unpickleProofStateFrom": "test/c.olean"}

{"tactic": "have t : Nat := 42", "proofState": 2}

{"pickleTo": "test/d.olean", "proofState": 3}

{"unpickleProofStateFrom": "test/d.olean"}

{"tactic": "exact t", "proofState": 5}
