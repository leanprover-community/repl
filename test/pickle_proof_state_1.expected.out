{"env": 0}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 18},
   "parentDecl": "f",
   "goals": ["⊢ Nat"],
   "endPos": {"line": 1, "column": 23}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 4},
   "endPos": {"line": 1, "column": 5},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals": ["⊢ Nat"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["⊢ Nat"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 3,
 "goals": ["t : Nat\n⊢ Nat"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 4,
 "goals": ["t : Nat\n⊢ Nat"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 5,
 "goals": ["t : Nat\n⊢ Nat"]}

{"proofStatus": "Completed", "proofState": 6, "goals": []}

